dev-tools/mage/build.go

```diff
@@ -46,6 +47,39 @@ type BuildArgs struct {
 	WinMetadata bool // Add resource metadata to Windows binaries (like add the version number to the .exe properties).
 }
 
+// buildTagRE is a regexp to match strings like "-tags=abcd"
+// but does not match "-tags= "
+var buildTagRE = regexp.MustCompile(`-tags=([\S]+)`)
+
+// ParseExtraFlags returns the ExtraFlags param where all flags that are go build tags are joined by a comma.
+//
+// For example if given -someflag=val1 -tags=buildtag1 -tags=buildtag2
+// It will return -someflag=val1 -tags=buildtag1,buildtag2
+func (b BuildArgs) ParseExtraFlags() []string {
```

Comment:
When building agentbeat without this parsing, it results in two `-tags=` values being sent to the `go build` command. one tag gets ignored; this resulted in the behaviour in my previous comment where I then had to adjust a runtime env var to get the binary in compliance.

Parsing the extra flags to join all `-tags=` flags is required in order to send `-tags=agentbeat,requirefips` which produced a binary that works as expected.

Reply:
How does this interact with the build tags for testing?  I'm thinking of "integration" as an example.  I'm assuming we will need tests that require both a testing build tag and the fips build tag.

https://github.com/elastic/beats/blob/1473ae916ab0da9d86ba07a48e10ba04128da574/dev-tools/mage/gotest.go#L297

Reply:
Thanks, I've altered `gotest.go` to join the tags with a comma to be consistent